app/src/main/java/com/github/k409/fitflow/ui/screens/goals/GoalCreation.kt

```diff
@@ -0,0 +1,180 @@
+                    } else {
+                        distanceOptionsWeek
+                    }
+
+                    DropdownMenu(
+                        options = distanceOptions,
+                        selectedOption = selectedDistance,
+                        label = "Select Distance",
+                        onOptionSelected = { selectedDistance = it },
+                    )
+
+                    InlineError(selectedDistance.isEmpty() && showInlineError)
+
+                    Spacer(modifier = Modifier.height(24.dp))
+                } else {
+                    NoValidGoalsMessage()
+                }
+            }
+
+            Button(onClick = {
+                if (selectedGoal.isNotEmpty() && selectedExercise.isNotEmpty() && selectedDistance.isNotEmpty()) {
+                    showConfirmationDialog = true
+                } else {
+                    showInlineError = true
+                }
+            }) {
+                Text("Create")
+            }
+
+            if (showConfirmationDialog) {
+                ConfirmationDialog(
+                    selectedGoal,
+                    selectedExercise,
+                    selectedDistance,
+                    onDismissRequest = { showConfirmationDialog = false },
+                    onConfirm = {
+                        goalsViewModel.submitGoalAsync(selectedGoal, selectedExercise, selectedDistance.filter { it.isDigit() || it == '.' }.toDoubleOrNull() ?: 0.0)
+                        showConfirmationDialog = false
+                        navController.navigate(NavRoutes.Goals.route)
+                    },
+                )
+            }
+        }
+    }
+}
+
+@Composable
+fun InlineError(show: Boolean) {
+    if (show) {
+        Text(
+            text = "Please select this field",
+            color = MaterialTheme.colorScheme.error,
+            style = MaterialTheme.typography.bodySmall,
+            modifier = Modifier.padding(start = 16.dp, top = 2.dp),
+        )
+    }
+}
+
+@Composable
+fun NoValidGoalsMessage() {
+    Text(
+        text = "No valid exercises are available for the selected goal period.",
+        color = MaterialTheme.colorScheme.error,
+        style = MaterialTheme.typography.bodyLarge,
+        modifier = Modifier.padding(16.dp),
+    )
+}
+
+@Composable
+fun ConfirmationDialog(
+    selectedGoal: String,
+    selectedExercise: String,
+    selectedDistance: String,
+    onDismissRequest: () -> Unit,
+    onConfirm: () -> Unit,
+) {
+    AlertDialog(
+        onDismissRequest = onDismissRequest,
+        title = { Text("Are you sure you want to create this goal?") },
+        text = { Text("$selectedGoal\n$selectedExercise\n$selectedDistance") },
+        confirmButton = {
+            Button(onClick = onConfirm) {
+                Text("Confirm")
+            }
+        },
+        dismissButton = {
+            Button(onClick = onDismissRequest) {
+                Text("Cancel")
+            }
+        },
+    )
+}
```

Comment:
I would suggest to remove this and instead use a very similar dialog ConfirmDialog(), defined in ui/common/Dialogs.kt file.